scripts/cache_vision_features.py

```diff
@@ -0,0 +1,58 @@
+import os
+import glob
+import argparse
+from tqdm import tqdm
+
+import torch
+from torchvision.datasets.folder import IMG_EXTENSIONS
+
+from allennlp.common.file_utils import TensorCache, cached_path
+from allennlp.data import DetectronImageLoader
+from allennlp.modules.vision import ResnetBackbone, FasterRcnnRegionDetector
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--image-dir", type=str, required=True)
+    parser.add_argument("--cache-dir", type=str, required=True)
+    parser.add_argument("--use-cuda", action="store_true", help="use GPU if one is available")
+    parser.add_argument("--batch-size", type=int, default=16)
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    os.makedirs(args.cache_dir, exist_ok=True)
+    features_cache = TensorCache(os.path.join(args.cache_dir, "features"))
+    coordinates_cache = TensorCache(os.path.join(args.cache_dir, "coordinates"))
+    image_paths = []
+    for extension in IMG_EXTENSIONS:
+        image_paths += list(glob.iglob(os.path.join(args.image_dir, "**", "*."+extension), recursive=True))
+    filtered_image_paths = []
+    image_keys = []
+    for path in image_paths:
+        key = os.path.basename(path)
+        if key in features_cache and key in coordinates_cache:
+            continue
```

Comment:
It's a bit of a nit, but can you delay this check until the second loop? It means you have to re-work the batching logic a bit, but it has several advantages:
1. It will start up much quicker when you have lots of images.
2. You can run multiple copies of this script at the same time, pointing at the same cache files. Also, you can run this at the same time as a training job, all pointing at the same files, and they will coordinate (almost) perfectly.